iron-list.html

```diff
@@ -395,7 +396,7 @@
       this._firstVisibleIndexVal = null;
 
       // random access
-      if (this._scrollBottom > this._physicalBottom || scrollTop < this._physicalTop) {
+      if (Math.abs(this._physicalBottom - this._scrollBottom) > this._physicalSize) {

```

Comment:
This logic doesn't look correct to me.

You need to enter random access mode when you know you can no longer flip items contiguously from top-to-bottom (or vice-versa), and will need to guess.  As long as the delta is less than the physicalSize, you should be able to flip enough items on average to meet your ratio on the other side.

I tried drawing a picture to confirm/illustrate:

![image](https://cloud.githubusercontent.com/assets/1354932/8758770/f2f75f38-2c9c-11e5-83a4-1a2333b7c690.png)

Green is the viewport, which moves from scrollTop1 to scrollTop2.  As long as the scroll delta is < the physicalSize, you can flip items contiguously (e.g., offset incremented by height of previously known item height) and at the limit, still maintain your previous overhang ratio.

So this seems more correct: `if (Math.abs(delta) > this._physicalSize)`